### PR TITLE
Docs: Update logit processors __call__ docs

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

PR done as part of https://github.com/huggingface/transformers/issues/24575

This PR polishes the inexistent `__call__` method docs for the logit processors (before this PR, only the base classes had docs).

### CODE REVIEW

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -30,17 +30,10 @@
 LOGITS_PROCESSOR_INPUTS_DOCSTRING = r"""
     Args:
         input_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`):
-            Indices of input sequence tokens in the vocabulary.
-
-            Indices can be obtained using [`BertTokenizer`]. See [`PreTrainedTokenizer.encode`] and
-            [`PreTrainedTokenizer.__call__`] for details.
-
-            [What are input IDs?](../glossary#input-ids)
+            Indices of input sequence tokens in the vocabulary. [What are input IDs?](../glossary#input-ids)
```
Comment (gante) [inline src/transformers/generation/logits_process.py:33]: `LogitsProcessor` is not directly used, but rather as part of `generate`. As such, the lines regarding how to obtain `input_ids` are not relevant here (which makes the docs more concise IMO :) )

File: src/transformers/generation/logits_process.py [hunk 2]
```diff
@@ -30,17 +30,10 @@
...
         scores (`torch.FloatTensor` of shape `(batch_size, config.vocab_size)`):
             Prediction scores of a language modeling head. These can be logits for each vocabulary when not using beam
             search or log softmax for each vocabulary token when using beam search
-        kwargs:
```
Comment (gante) [inline src/transformers/generation/logits_process.py:42]: having extra `kwargs` is the exception -- this bit was moved to the docs of the classes that need it

File: src/transformers/generation/logits_process.py [hunk 3]
```diff
@@ -77,8 +68,22 @@ class LogitsProcessorList(list):
...
-    @add_start_docstrings(LOGITS_PROCESSOR_INPUTS_DOCSTRING)
     def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> torch.FloatTensor:
+        r"""
+        Args:
+            input_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`):
+                Indices of input sequence tokens in the vocabulary. [What are input IDs?](../glossary#input-ids)
+            scores (`torch.FloatTensor` of shape `(batch_size, config.vocab_size)`):
+                Prediction scores of a language modeling head. These can be logits for each vocabulary when not using
+                beam search or log softmax for each vocabulary token when using beam search
+            kwargs (`Dict[str, Any]`):
```
Comment (gante) [inline src/transformers/generation/logits_process.py:79]:
```suggestion
            kwargs (`Dict[str, Any]`, *optional*):
```

**Review by amyeroberts — APPROVED** [review]
Thanks for updating!

**Pull request merged**